pkg/kubectl/cmd/rollingupdate.go

```diff
@@ -257,14 +255,13 @@ func RunRollingUpdate(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, arg
 		updateCleanupPolicy = kubectl.RenameRollingUpdateCleanupPolicy
 	}
 	config := &kubectl.RollingUpdaterConfig{
-		Out:            out,
-		OldRc:          oldRc,
-		NewRc:          newRc,
-		UpdatePeriod:   period,
-		Interval:       interval,
-		Timeout:        timeout,
-		CleanupPolicy:  updateCleanupPolicy,
-		UpdateAcceptor: kubectl.DefaultUpdateAcceptor,
+		Out:           out,
+		OldRc:         oldRc,
+		NewRc:         newRc,
+		UpdatePeriod:  period,
+		Interval:      interval,
+		Timeout:       timeout,
+		CleanupPolicy: updateCleanupPolicy,
```

Comment:
Maybe just add a Boolean "useReadinessAsAcceptance".  I don't want to
invent a generic interface if readiness is what we want in the long run.

On Jul 31, 2015, at 10:38 AM, Dan Mace <notifications@github.com> wrote:

In pkg/kubectl/cmd/rollingupdate.go
<https://github.com/GoogleCloudPlatform/kubernetes/pull/11942#discussion_r35979259>
:

> @@ -257,14 +255,13 @@ func RunRollingUpdate(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, arg
>  		updateCleanupPolicy = kubectl.RenameRollingUpdateCleanupPolicy
>  	}
>  	config := &kubectl.RollingUpdaterConfig{
> -		Out:            out,
> -		OldRc:          oldRc,
> -		NewRc:          newRc,
> -		UpdatePeriod:   period,
> -		Interval:       interval,
> -		Timeout:        timeout,
> -		CleanupPolicy:  updateCleanupPolicy,
> -		UpdateAcceptor: kubectl.DefaultUpdateAcceptor,
> +		Out:           out,
> +		OldRc:         oldRc,
> +		NewRc:         newRc,
> +		UpdatePeriod:  period,
> +		Interval:      interval,
> +		Timeout:       timeout,
> +		CleanupPolicy: updateCleanupPolicy,

So should we go ahead and make the availability type public and make it a
field on RollingUpdaterConfig? The default would still be "created".

—
Reply to this email directly or view it on GitHub
<https://github.com/GoogleCloudPlatform/kubernetes/pull/11942/files#r35979259>
.